### ISSUE TITLE
LSP: Handle language servers that return vim.NIL

### ISSUE THREAD

**Issue by PhilT** [neovim/neovim#12824]
Some language servers servers are returning `vim.NIL` in their json instead of `nil`. This is causing various errors as reported in https://github.com/neovim/nvim-lspconfig/issues/304. 

Here is an example from the F# Language Server. 

```
{                                                             
  additionalTextEdits = {},                       
  command = vim.NIL,
  commitCharacters = {},                          
  data = {                                                  
    FullName = "System.IO.DirectoryInfo"
  },                                                     
  detail = "System.IO.DirectoryInfo",
  documentation = vim.NIL,
  filterText = vim.NIL,   
  insertText = "DirectoryInfo",
  insertTextFormat = vim.NIL,
  kind = 7,                       
  label = "DirectoryInfo",
  sortText = vim.NIL,
  textEdit = vim.NIL
}
```

A quick fix would be to add `vim.NIL` checks to the existing `nil` checks in `vim/lsp/util.lua`. I have a WIP of this which appears to fix the problem. However, I'm wondering if a better solution would be to replace `vim.NIL` with `nil` rather than do extra checks everywhere. However, I don't know enough about the reason `vim.NIL` is used to be sure this is the correct solution. I'd be happy to implement a PR, given some direction.

**Comment by mjlbach** [comment]
Wasn't this closed by #13174?

**Comment by PhilT** [comment]
yep
